smtpserver/server.go

```diff
@@ -1654,6 +1658,53 @@ func (c *conn) cmdRcpt(p *parser) {
 	c.bwritecodeline(smtp.C250Completed, smtp.SeAddr1Other0, "now on the list", nil)
 }
 
+// ../rfc/6531:497
+func (c *conn) isSmtpUtf8Required(part *message.Part) bool {
+	hasNonASCII := func(r io.Reader) bool {
```

Comment:
I found two other functions similar to `hasNonASCII`:
* `isASCII` in message/compose.go:169
* `isASCII` in publicsuffix/list.go:162

Maybe we should refactor those functions and declare it only once ? (but where?)

Reply:
we could merge it, but not sure if worth the trouble. indeed due to finding a place to put it (i think moxio would be a candidate, it's most like a "misc util" package. also, publicsuffix is what i consider a "leaf package", not great to include a dependency on a misc package. and then there is the required error handling that comes with reading from an io.Reader. since it's a small function, may be better to leave duplicates.